packages/calypso-e2e/src/lib/components/editor-gutenberg-component.ts

```diff
@@ -0,0 +1,201 @@
+import { Page, FrameLocator, ElementHandle } from 'playwright';
+
+const editorPane = 'div.edit-post-visual-editor__content-area';
+const selectors = {
+	// Title
+	title: '.editor-post-title__input',
+
+	// Editor body
+	emptyBlock: 'div.block-editor-default-block-appender', // When editor is in a 'resting' state, without a selected block.
+	paragraphBlock: ( empty: boolean ) => `p[data-title="Paragraph"][data-empty="${ empty }"]`,
+	blockWarning: '.block-editor-warning',
+
+	// Block Search
+	blockSearchInput: '.block-editor-inserter__search input[type="search"]',
+	blockResultItem: ( name: string ) =>
+		`.block-editor-block-types-list__list-item span:text("${ name }")`,
+	patternResultItem: ( name: string ) => `div[aria-label="${ name }"]`,
+};
+
+/**
+ * Represents an instance of the Gutenberg Block Editor as loaded on
+ * WordPress.com.
+ */
+export class EditorGutenbergComponent {
+	private page: Page;
+	private frameLocator: FrameLocator;
+
+	/**
+	 * Constructs an instance of the component.
+	 *
+	 * @param {Page} page The underlying page.
+	 * @param {FrameLocator} frameLocator Locator of the editor iframe.
+	 */
+	constructor( page: Page, frameLocator: FrameLocator ) {
+		this.page = page;
+		this.frameLocator = frameLocator;
+	}
+
+	/**
+	 * Resets the selected block.
+	 *
+	 * The Gutenberg block-based editor 'remembers' what block was last
+	 * selected. This behavior impacts the block options that are shown
+	 * in the block inserter.
+	 *
+	 * For instance, if a Contact Form block is currently selected, the
+	 * block inserter will display a filtered set of blocks that are
+	 * permitted to be inserted within the parent Contact Form block.
+	 */
+	async resetSelectedBlock(): Promise< void > {
+		const locator = this.frameLocator.locator( selectors.title );
+		await locator.click();
+	}
+
+	/* Title block */
+
+	/**
+	 * Enters the text into the title block and verifies the result.
+	 *
+	 * @param {string} title Text to be used as the title.
+	 */
+	async enterTitle( title: string ): Promise< void > {
+		const locator = this.frameLocator.locator( selectors.title );
+		await locator.fill( title );
+	}
+
+	/**
+	 * Returns the content of the Page/Post title.
+	 *
+	 * @returns {Promise<string>} String containing contents of the title.
+	 */
+	async getTitle(): Promise< string > {
+		const locator = this.frameLocator.locator( selectors.title );
+		return await locator.innerText();
+	}
+
+	/* Paragraph block shortcuts */
+
+	/**
+	 * Enters text into the paragraph block(s) and verifies the result.
+	 *
+	 * Note that this method of text entry does not explicitly use the
+	 * ParagraphBlock construct. This is due to text entry being a high
+	 * frequency use case.
+	 *
+	 * @param {string} textArray Array of text to be entered. Each entry is
+	 * treated as individual Paragraph blocks.
+	 */
+	async enterText( textArray: string[] ): Promise< void > {
```

Comment:
Definitely not required, but is there any reason not to have this signature match the signature on the EditorPage, and just accept  a `string` and do the `\n` splitting here? Or maybe alternatively have this method here take either a `string` or a `string[]`, and if a `string` is provided, do the splitting on `\n`? I figured it might be helpful to consolidate any potential logic in one place, here in the subcomponent.

Reply:
I've modified this method to return a single string.